### PR TITLE
Refactor inplaceOnlyPodUpdater plugin

### PR DESCRIPTION
**Pull request by ColdsteelRail**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [ ] Y 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [ ] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/controllers/collaset/synccontrol/update.go [hunk 1]
```diff
@@ -324,6 +324,7 @@ func (o orderByDefault) Less(i, j int) bool {
 }
 
 type PodUpdater interface {
+	SetupPodUpdater(context.Context, client.Client, *appsv1alpha1.CollaSet, podcontrol.Interface, record.EventRecorder)
```
Comment (wu8685) [inline pkg/controllers/collaset/synccontrol/update.go:327]: Setup is fine.

File: pkg/controllers/collaset/synccontrol/update.go [hunk 2]
```diff
@@ -333,17 +334,25 @@ type PodUpdater interface {
 }
 
 type GenericPodUpdater struct {
-	collaSet   *appsv1alpha1.CollaSet
-	ctx        context.Context
-	podControl podcontrol.Interface
-	recorder   record.EventRecorder
+	*appsv1alpha1.CollaSet
+	context.Context
```
Comment (wu8685) [inline pkg/controllers/collaset/synccontrol/update.go:338]: Maybe we can remove `context` this time.
In most of time, context is passed when method is called. It should not be carried as a object field.

Comment (ColdsteelRail) [reply pkg/controllers/collaset/synccontrol/update.go]: reasonable, ctx is removed from object filed and put into func params

**Review by wu8685 — APPROVED** [review]
lgtm

**Pull request merged**